### ISSUE TITLE
Question about random_walk part

### ISSUE THREAD

**Issue by Six-walnuts-Lu** [LeonHLJ/RSKP#8]
Thank you very much for your work.
```
def random_walk(x, y, w):
    x_norm = calculate_l1_norm(x)
    y_norm = calculate_l1_norm(y)
    eye_x = torch.eye(x.size(1)).float().to(x.device)
    latent_z = F.softmax(torch.einsum('nkd,ntd->nkt', [y_norm, x_norm]) * 5.0, 1)
    norm_latent_z = latent_z / (latent_z.sum(dim=-1, keepdim=True) + 1e-9)
    affinity_mat = torch.einsum('nkt,nkd->ntd', [latent_z, norm_latent_z])
    mat_inv_x, _ = torch.solve(eye_x, eye_x - (w ** 2) * affinity_mat)
    y2x_sum_x = w * torch.einsum('nkt,nkd->ntd', [latent_z, y]) + x
    refined_x = (1 - w) * torch.einsum('ntk,nkd->ntd', [mat_inv_x, y2x_sum_x])    
    return refined_x
```
I don't understand the logic of this part. Is there any paper about bipartite random walk (BiRW)? Or please elaborate on the logic of this part. Thank you very much.

**Comment by LeonHLJ** [comment]
> Thank you very much for your work.
> 
> ```
> def random_walk(x, y, w):
>     x_norm = calculate_l1_norm(x)
>     y_norm = calculate_l1_norm(y)
>     eye_x = torch.eye(x.size(1)).float().to(x.device)
>     latent_z = F.softmax(torch.einsum('nkd,ntd->nkt', [y_norm, x_norm]) * 5.0, 1)
>     norm_latent_z = latent_z / (latent_z.sum(dim=-1, keepdim=True) + 1e-9)
>     affinity_mat = torch.einsum('nkt,nkd->ntd', [latent_z, norm_latent_z])
>     mat_inv_x, _ = torch.solve(eye_x, eye_x - (w ** 2) * affinity_mat)
>     y2x_sum_x = w * torch.einsum('nkt,nkd->ntd', [latent_z, y]) + x
>     refined_x = (1 - w) * torch.einsum('ntk,nkd->ntd', [mat_inv_x, y2x_sum_x])    
>     return refined_x
> ```
> 
> I don't understand the logic of this part. Is there any paper about bipartite random walk (BiRW)? Or please elaborate on the logic of this part. Thank you very much.

You can first refer to "Shen, Yantao, et al. Deep group-shuffling random walk for person re-identification. CVPR 2018." for details of the random walk. Our method is a further step on the used random walk in this paper with a bipartite propagation process. If you have any questions, please feel free to contact me.
